engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneDelegate.m

```diff
@@ -16,20 +16,25 @@ - (void)scene:(UIScene*)scene
     willConnectToSession:(UISceneSession*)session
                  options:(UISceneConnectionOptions*)connectionOptions {
   NSObject<UIApplicationDelegate>* appDelegate = FlutterSharedApplication.application.delegate;
-  if (appDelegate.window.rootViewController) {
+  if ([appDelegate respondsToSelector:@selector(window)] && appDelegate.window.rootViewController) {
```

Comment:
are you able to repro the crash? `UIApplicationDelegate` does have this `window` property (it could be nil but shouldn't crash).

Reply:
Thanks for confirming! 

Very interesting: https://developer.apple.com/documentation/uikit/uiapplicationdelegate/window?language=objc

> If you want to provide a custom window for your app, you must implement the getter method of this property and use it to create and return your custom window.

Sounds like we should have added this `window` property in our guide. Though with SceneDelegate API, all `window`/`vc` related logic should be moved.